src/Microsoft.AspNet.Mvc.Xml/Formatters/XmlOutputFormatter.cs

```diff
@@ -37,16 +37,22 @@ public XmlOutputFormatter([NotNull] XmlWriterSettings xmlWriterSettings)
         /// <returns>The type of the object to be serialized.</returns>
         protected virtual Type GetSerializableType(Type declaredType, Type runtimeType)
         {
-            if (declaredType == null ||
-                declaredType == typeof(object))
+            Type type = null;
+
+            if (declaredType == null || declaredType == typeof(object))
             {
                 if (runtimeType != null)
                 {
-                    return runtimeType;
+                    type = runtimeType;
                 }
             }
 
-            return declaredType;
+            if (type == null)
+            {
+                type = declaredType;
+            }
+
+            return SerializableErrorWrapper.CreateSerializableType(type);
```

Comment:
this needs to change for a general wrapper. this can happen along with the IEnumerable support PR though if that's your intention